src/protocol/decoder.js

```diff
@@ -57,11 +57,22 @@ module.exports = class Decoder {
   }
 
   readInt64() {
-    const lowBits = this.buffer.readInt32BE(this.offset + 4)
-    const highBits = this.buffer.readInt32BE(this.offset)
-    const value = new Long(lowBits, highBits)
+    const first = this.buffer[this.offset]
+    const last = this.buffer[this.offset + 7]
+
+    const low =
+      (first << 24) + // Overflow
```

Comment:
I tried to change this to use left shifting everywhere, instead of the exponentiation, since in my mind `x << y` is the same as `x * 2 ** y`, but it seems that is not the case, because when I tried I always got weird results.

Reply:
One thing I still don't really understand is that in every other implementation I've seen, reading a BigInt64 from a buffer first checks if there are at least 8 bytes available (the length of a 64 bit integer), but then it only advances the offset by 6 bytes (where I'm doing `offset + n` the other implementations do `++offset`). That would lead to incorrect results, so I don't understand why they do that or why it works for them.